storage/range_raftstorage.go

```diff
@@ -334,11 +344,24 @@ func (r *Range) ApplySnapshot(snap raftpb.Snapshot) error {
 		return nil
 	}
 
+	// Read the updated range descriptor.
+	var desc proto.RangeDescriptor
+	// KV[0] acts as metadata which stores RangeDescriptor.
+	// see comments in Snapshot().
+	descKey := keys.RaftRangeDescriptorKey(r.Desc().RaftID)
+	if !descKey.Equal(snapData.KV[0].Key) {
+		return nil
```

Comment:
This should return an error, not nil. (same for all the others in this function; I'm pretty sure that all the `if err != nil { return nil }` blocks are just mistakes. If we want to ignore an error we should at least log that it happened or comment about why).